plugins/filter_aws/aws.c

```diff
@@ -427,8 +429,113 @@ static int get_ec2_tag_values(struct flb_filter_aws *ctx)
     return 0;
 }
 
+static int tag_is_present_in_list(struct flb_filter_aws *ctx, flb_sds_t tag,
+        const char *tags)
+{
+    const char *tag_start;
+    const char *tag_end;
+    const char *tag_sep_before;
+    const char *tags_start;
+    const char *tag_sep_after;
+    const char *tags_end;
+
+    tag_start = strstr(tags, tag);
+    if (!tag_start) {
+        return FLB_FALSE;
+    }
+    tag_end = tag_start + strlen(tag) - 1;
+
+    /* if substring exists it not necessarily implies that tag is included */
+    /* tag name must be exactly equal to the value between separators ',' */
+    /* f.e. without additional conditions, substring would accept Name for a,Namee,b */
+
+    tags_start = tags;
+    tags_end = tags + strlen(tags) - 1;
+
+    /* calculate pointers to characters directly before/after tag */
+    /* if tag is in the middle, then these pointers should be ',' */
+    tag_sep_before = (char*)(tag_start - 1);
+    tag_sep_after = (char*)(tag_end + 1);
```

Comment:
won't this run over the beginning and end of the char array???

Or do I not understand this code... it looks to me like tags is some char array from the config map option and we initialize start and end to the first and last char.

Reply:
It calculates the potential pointers, but we don't access the memory there, so it's safe.

Yes, `tags` is the char array from the configuration. It's represented by tag values separated by a semicolon. `tag` is the value taken from the metadata server, which represents a single EC2 tag.

In these lines we calculate the potential position of the semicolon. It may lead to the unsafe parts of the memory, so if this is the case, we correct it to NULL right after.

Reply:
This is an old implementation, which no longer exists, so I am going to resolve the conversation.